iothub_client/src/blob.c

```diff
@@ -48,48 +148,52 @@ BLOB_RESULT Blob_UploadBlock(
     }
     else
     {
-        char temp[7]; /*this will contain 000000... 049999*/
-        if (sprintf(temp, "%6u", (unsigned int)blockID) != 6) /*produces 000000... 049999*/
+        char blockIdString[7]; /*this will contain 000000... 049999*/
```

Comment:
Why 7? Can we have a #define or a named const?

Reply:
Same for the 6. (Pointing to the Blob spec would be a good idea since it seems to be their requirement to have numbers formatted this way.)

Reply:
Done. This is also how it was implemented before. The refactoring done did not target every single aspect of the previous code.